src/Microsoft.AspNet.Authentication/RemoteAuthenticationHandler.cs

```diff
@@ -36,8 +36,7 @@ public abstract class RemoteAuthenticationHandler<TOptions> : AuthenticationHand
                     return false;
                 }
 
-                Context.Response.StatusCode = 500;
-                return true;
+                throw errorContext.Error;
```

Comment:
Not usually no, but the certainly can be. We don't want to loose that information if it's there.